noxfile.py

```diff
@@ -0,0 +1,42 @@
+from pathlib import Path
+
+import nox
+
+
+@nox.session(name="lint")
+def lint(session):
+    # run linters
+    session.install("ruff")
+    session.run("ruff", "check", "--ignore", "D")
+
+
+@nox.session(name="tests", python=["3.10", "3.11", "3.12"])
+def tests(session):
+    # run tests
+    session.install("pytest")
+    # Install pytest-cov for coverage reporting
+    session.install("pytest-cov")
+    # Install dependencies listed in pyproject.toml
+    session.install(
```

Comment:
instead of all this install and pythonpath manipulation, do `session.install(".[dev]")` (see [here](https://learn.scientific-python.org/development/guides/tasks/#introduction))